shimmy/openspiel_wrapper.py

```diff
@@ -0,0 +1,266 @@
+"""Wrapper to convert a openspiel environment into a pettingzoo compatible environment."""
+
+import functools
+from typing import Dict, Optional
+
+import numpy as np
+import pettingzoo as pz
+import pyspiel
+from gymnasium import spaces
+from gymnasium.utils import seeding
+from pettingzoo.utils.env import AgentID
+
+
+class OpenspielWrapper(pz.AECEnv):
+    """Wrapper that converts a openspiel environment into a pettingzoo environment."""
+
+    metadata = {"render_modes": [None]}
+
+    def __init__(
+        self,
+        game: pyspiel.Game,
+        render_mode: Optional[str],
+    ):
+        """Wrapper that converts a openspiel environment into a pettingzoo environment.
+
+        Args:
+            game (pyspiel.Game): game
+            render_mode (Optional[str]): render_mode
+        """
+        self.game = game
+        self.possible_agents = [
+            "player_" + str(r) for r in range(self.game.num_players())
+        ]
+        self.agent_id_name_mapping = dict(
+            zip(range(self.game.num_players()), self.possible_agents)
+        )
+        self.agent_name_id_mapping = dict(
+            zip(self.possible_agents, range(self.game.num_players()))
+        )
+
+        self.render_mode = render_mode
+
+    @functools.lru_cache(maxsize=None)
+    def observation_space(self, agent: AgentID):
+        """observation_space.
+
+        Args:
+            agent (AgentID): agent
+        """
+        try:
+            return spaces.Box(
+                low=-np.inf, high=np.inf, shape=self.game.observation_tensor_shape()
+            )
+        except pyspiel.SpielError as e:
+            raise NotImplementedError(f"{str(e)[:-1]} for {self.game}.")
+
+    @functools.lru_cache(maxsize=None)
+    def action_space(self, agent: AgentID):
+        """action_space.
+
+        Args:
+            agent (AgentID): agent
+        """
+        try:
+            return spaces.Discrete(self.game.num_distinct_actions())
+        except pyspiel.SpielError as e:
+            raise NotImplementedError(f"{str(e)[:-1]} for {self.game}.")
+
+    def render(self):
+        """render."""
+        raise NotImplementedError("No render available for openspiel.")
+
+    def observe(self, agent: AgentID):
+        """observe.
+
+        Args:
+            agent (AgentID): agent
+        """
+        return np.array(self.observations[agent])
+
+    def close(self):
+        """close."""
+        pass
+
+    def reset(
+        self,
+        seed: Optional[int] = None,
+        return_info: Optional[bool] = False,
```

Comment:
Does PZ still have a `return_info` parameter? Will this not be removed soon

Reply:
Good point, removed it.

Reply:
Yes, api test won't pass without it